src/Assetic/Filter/CoffeeScriptFilter.php

```diff
@@ -64,10 +59,21 @@ public function filterLoad(AssetInterface $asset)
             $pb->add('--no-header');
         }
 
-        $pb->add($input);
+        // compile to JavaScript, otherwise coffee will run the script
+        $pb->add('--compile');
+
+        // we want it to print the output rather than run it
+        $pb->add('--print');
+
+        // read content from standard input
+        $pb->add("--stdio");
```

Comment:
Using STDIN and STDOUT has caused issues in the past, which is why Assetic uses temporary files whenever possible. I think the issues were mostly on Windows.